From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: boringwolf <boringwolf@leandev.io>
Date: Wed, 8 May 2024 18:14:53 +0800
Subject: [PATCH] =?UTF-8?q?=E7=94=A8=20HashCode=20=E7=A2=BA=E8=AA=8D?=
 =?UTF-8?q?=E6=9B=B4=E6=96=B0=E8=88=87=E5=90=A6?=
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit


diff --git a/src/main/java/io/papermc/paper/plugin/provider/source/FileProviderSource.java b/src/main/java/io/papermc/paper/plugin/provider/source/FileProviderSource.java
index 4656376505dac62bf8778eae8d11566c1a1a337d..fa0a65c3666166e6d05ba35d028205d1e3c0d973 100644
--- a/src/main/java/io/papermc/paper/plugin/provider/source/FileProviderSource.java
+++ b/src/main/java/io/papermc/paper/plugin/provider/source/FileProviderSource.java
@@ -1,5 +1,6 @@
 package io.papermc.paper.plugin.provider.source;
 
+import com.google.common.hash.HashCode;
 import com.mojang.logging.LogUtils;
 import io.papermc.paper.plugin.PluginInitializerManager;
 import io.papermc.paper.plugin.entrypoint.EntrypointHandler;
@@ -103,8 +104,13 @@ public class FileProviderSource implements ProviderSource<Path, Path> {
         try {
             String pluginName = this.getPluginName(file);
             String pluginVersion = this.getPluginVersion(file);
-            UpdateFileVisitor visitor = new UpdateFileVisitor(pluginName, pluginVersion);
+            HashCode pluginHash = HashCode.fromBytes(Files.readAllBytes(file));
+            UpdateFileVisitor visitor = new UpdateFileVisitor(pluginName);
             Files.walkFileTree(updateDirectory, Set.of(), 1, visitor);
+
+            if (pluginHash.equals(updateHash)) {
+                return file;
+            }
             if (visitor.getValidPlugin() != null) {
                 Path updateLocation = visitor.getValidPlugin();
                 String updatePluginVersion = this.getPluginVersion(updateLocation);
@@ -152,13 +158,11 @@ public class FileProviderSource implements ProviderSource<Path, Path> {
     private class UpdateFileVisitor implements FileVisitor<Path> {
 
         private final String targetName;
-        private final String targetVersion;
         @Nullable
         private Path validPlugin;
 
-        private UpdateFileVisitor(String targetName, String targetVersion) {
+        private UpdateFileVisitor(String targetName) {
             this.targetName = targetName;
-            this.targetVersion = targetVersion;
         }
 
         @Override
@@ -170,8 +174,7 @@ public class FileProviderSource implements ProviderSource<Path, Path> {
         public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
             try {
                 String updatePluginName = FileProviderSource.this.getPluginName(file);
-                String updatePluginVersion = FileProviderSource.this.getPluginVersion(file);
-                if (this.targetName.equals(updatePluginName) && !this.targetVersion.equals(updatePluginVersion)) {
+                if (this.targetName.equals(updatePluginName)) {
                     this.validPlugin = file;
                     return FileVisitResult.TERMINATE;
                 }
