From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: boringwolf <boringwolf@leandev.io>
Date: Wed, 8 May 2024 19:59:03 +0800
Subject: [PATCH] =?UTF-8?q?=E6=96=B0=E5=A2=9E=20Debug=20=E8=A8=8A=E6=81=AF?=
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit


diff --git a/src/main/java/io/papermc/paper/plugin/provider/source/FileProviderSource.java b/src/main/java/io/papermc/paper/plugin/provider/source/FileProviderSource.java
index fa0a65c3666166e6d05ba35d028205d1e3c0d973..51476dcc9be94e13a14d588443633e967af33365 100644
--- a/src/main/java/io/papermc/paper/plugin/provider/source/FileProviderSource.java
+++ b/src/main/java/io/papermc/paper/plugin/provider/source/FileProviderSource.java
@@ -108,6 +108,8 @@ public class FileProviderSource implements ProviderSource<Path, Path> {
             UpdateFileVisitor visitor = new UpdateFileVisitor(pluginName);
             Files.walkFileTree(updateDirectory, Set.of(), 1, visitor);
 
+            LOGGER.debug("Checking for plugin update: {} ({} -> {})", pluginName, pluginHash, updateHash);
+
             if (pluginHash.equals(updateHash)) {
                 return file;
             }
@@ -127,6 +129,7 @@ public class FileProviderSource implements ProviderSource<Path, Path> {
                 LOGGER.info("Updated plugin {} from version {} to version {}", pluginName, pluginVersion, updatePluginVersion);
                 return newName.toPath();
             }
+            LOGGER.debug("No update found for plugin {}", pluginName);
         } catch (Exception e) {
             throw new InvalidPluginException(e);
         }
