From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: boringwolf <boringwolf@leandev.io>
Date: Wed, 8 May 2024 20:11:58 +0800
Subject: [PATCH] debug to info


diff --git a/src/main/java/io/papermc/paper/plugin/provider/source/FileProviderSource.java b/src/main/java/io/papermc/paper/plugin/provider/source/FileProviderSource.java
index 51476dcc9be94e13a14d588443633e967af33365..e18399b800725958c6eb7a2b5cce10794be48980 100644
--- a/src/main/java/io/papermc/paper/plugin/provider/source/FileProviderSource.java
+++ b/src/main/java/io/papermc/paper/plugin/provider/source/FileProviderSource.java
@@ -108,7 +108,7 @@ public class FileProviderSource implements ProviderSource<Path, Path> {
             UpdateFileVisitor visitor = new UpdateFileVisitor(pluginName);
             Files.walkFileTree(updateDirectory, Set.of(), 1, visitor);
 
-            LOGGER.debug("Checking for plugin update: {} ({} -> {})", pluginName, pluginHash, updateHash);
+            LOGGER.info("Checking for plugin update: {} ({} -> {})", pluginName, pluginHash, updateHash);
 
             if (pluginHash.equals(updateHash)) {
                 return file;
@@ -129,7 +129,7 @@ public class FileProviderSource implements ProviderSource<Path, Path> {
                 LOGGER.info("Updated plugin {} from version {} to version {}", pluginName, pluginVersion, updatePluginVersion);
                 return newName.toPath();
             }
-            LOGGER.debug("No update found for plugin {}", pluginName);
+            LOGGER.info("No update found for plugin {}", pluginName);
         } catch (Exception e) {
             throw new InvalidPluginException(e);
         }
