From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: boringwolf <boringwolf@leandev.io>
Date: Wed, 8 May 2024 20:17:07 +0800
Subject: [PATCH] fix


diff --git a/src/main/java/io/papermc/paper/plugin/provider/source/FileProviderSource.java b/src/main/java/io/papermc/paper/plugin/provider/source/FileProviderSource.java
index 42577887352b225f0fe8fcf9fb7b8e17606a7061..4d29aee12d3995441735cfa85432c6d60be88f2f 100644
--- a/src/main/java/io/papermc/paper/plugin/provider/source/FileProviderSource.java
+++ b/src/main/java/io/papermc/paper/plugin/provider/source/FileProviderSource.java
@@ -100,7 +100,7 @@ public class FileProviderSource implements ProviderSource<Path, Path> {
                 Path updateLocation = visitor.getValidPlugin();
                 HashCode updateHash = HashCode.fromBytes(Files.readAllBytes(updateLocation));
 
-                LOGGER.info("Checking for plugin update: {} ({} -> {})", pluginName, pluginHash, updateHash);
+                LOGGER.info("Checking for plugin update: {} ({})", pluginName, pluginHash.equals(updateHash));
 
                 if (pluginHash.equals(updateHash)) {
                     return file;
