From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: boringwolf <boringwolf@leandev.io>
Date: Wed, 8 May 2024 20:17:07 +0800
Subject: [PATCH] fix


diff --git a/src/main/java/io/papermc/paper/plugin/provider/source/FileProviderSource.java b/src/main/java/io/papermc/paper/plugin/provider/source/FileProviderSource.java
index e18399b800725958c6eb7a2b5cce10794be48980..598ba7d6133531448b3135af08feee3072ca4dd7 100644
--- a/src/main/java/io/papermc/paper/plugin/provider/source/FileProviderSource.java
+++ b/src/main/java/io/papermc/paper/plugin/provider/source/FileProviderSource.java
@@ -108,7 +108,7 @@ public class FileProviderSource implements ProviderSource<Path, Path> {
             UpdateFileVisitor visitor = new UpdateFileVisitor(pluginName);
             Files.walkFileTree(updateDirectory, Set.of(), 1, visitor);
 
-            LOGGER.info("Checking for plugin update: {} ({} -> {})", pluginName, pluginHash, updateHash);
+            LOGGER.info("Checking for plugin update: {} ({})", pluginName, pluginHash.equals(updateHash));
 
             if (pluginHash.equals(updateHash)) {
                 return file;
